cmake/linker/ld/target.cmake

```
@@ -79,8 +79,14 @@ endmacro()
 
 # Force symbols to be entered in the output file as undefined symbols
 function(toolchain_ld_force_undefined_symbols)
+  if (CONFIG_ARCH_POSIX AND ${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Darwin")
+    set(symbol_prefix "_")
+  else()
+    set(symbol_prefix "")
+  endif()
+
   foreach(symbol ${ARGN})
-    zephyr_link_libraries(${LINKERFLAGPREFIX},-u,${symbol})
+    zephyr_link_libraries(${LINKERFLAGPREFIX},-u,${symbol_prefix}${symbol})
```

This should be described through a linker property, for example:
```
set_linker_property(TARGET linker APPEND PROPERTY symbol_prefix _)
```
inside:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/linker/ld/host-gcc/linker_flags.cmake
(again, we need to extend current design to take host system into consideration)

Then this code would simply become:
```suggestion
    zephyr_link_libraries(${LINKERFLAGPREFIX},-u,$<TARGET_PROPERTY:linker,symbol_prefix>${symbol})
```

Note: https://github.com/zephyrproject-rtos/zephyr/pull/24851 focused on compiler, we still need to cleanup of linker functions like `toolchain_ld_force_undefined_symbols` in similar way.

---

```
set_linker_property(TARGET linker APPEND PROPERTY symbol_prefix _)
```
results in a CMake error
```
Unknown CMake command "set_linker_property".
```